nntrainer/layers/common_properties.cpp

```diff
@@ -220,17 +220,47 @@ bool Padding1D::isValid(const std::string &v) const {
   return false;
 }
 
-std::array<unsigned int, 2> Padding1D::compute(const TensorDim &input,
-                                               const TensorDim &kernel,
-                                               const unsigned int &strides) {
+std::array<unsigned int, 2> Padding1D::compute(const TensorDim &input_dim,
+                                               const unsigned int &kernel,
+                                               const unsigned int &stride,
+                                               const unsigned int &dilation) {
   auto &padding_repr = get(); /// padding representation
 
+  auto calculate_padding = [](unsigned input, unsigned kernel, unsigned stride,
+                              unsigned dilation) {
+    /// ceil(input / stride)
+    unsigned int eff_kernel = (kernel - 1) * dilation + 1;
+    auto out = (input + stride - 1) / stride;
+    auto req_input = (out - 1) * stride + eff_kernel;
+    return req_input >= input ? req_input - input : 0;
+  };
+
   if (istrequal(padding_repr, "valid")) {
     return {0, 0};
+  } else if (istrequal(padding_repr, "same")) {
+
+    auto pad_horizontal =
+      calculate_padding(input_dim.width(), kernel, stride, dilation);
+
+    auto pad_left = pad_horizontal / 2;
+
+    return {pad_left, pad_horizontal - pad_left};
```

Comment:
pad_horizontal - pad_left = pad_horizontal - (pad_horizontal / 2) = pad_horizontal / 2 = pad_left
so,,,
return {pad_left, pad_left} ??

Reply:
That's right if the pad_horizontal is even number.
If the pad_horizontal is odd number, it will return {pad, left, pad_left + 1}

Reply:
thanks for checking.